### PR TITLE
Only scroll step when NumberField has focus

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/1602

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/numberfield/src/useNumberField.ts [hunk 1]
```diff
@@ -20,14 +20,14 @@ import {
   useCallback,
   useEffect,
   useMemo,
-  useRef
+  useRef, useState
```
Comment (LFDanLu) [inline packages/@react-aria/numberfield/src/useNumberField.ts:23]: nit
```suggestion
  useRef, 
  useState
```

**Review by LFDanLu — DISMISSED** [review]
LGTM, the useScrollWheel firing add and removeEventListener on every increment/decrement/hover threw me off (and still feels weird) but seems to have been from a older PR so ok w/ me

**Review by LFDanLu — APPROVED** [review]

File: packages/@react-aria/numberfield/src/useNumberField.ts [hunk 2]
```diff
@@ -72,7 +73,8 @@ export function useNumberField(props: AriaNumberFieldProps, state: NumberFieldSt
     autoFocus,
     validationState,
     label,
-    formatOptions
+    formatOptions,
+    disableScroll
```
Comment (devongovett) [inline packages/@react-aria/numberfield/src/useNumberField.ts:77]: Do we really need an option for this or is disabling when focused enough?

Comment (snowystinger) [reply packages/@react-aria/numberfield/src/useNumberField.ts:77]: you mean disabling when not focused?
it came from your comment https://github.com/adobe/react-spectrum/issues/1602#issuecomment-781339005
so I think you're saying you now think that completely disabling it not a needed feature

File: packages/@react-aria/numberfield/src/useNumberField.ts [hunk 3]
```diff
@@ -384,13 +390,18 @@ export function useNumberField(props: AriaNumberFieldProps, state: NumberFieldSt
 }
 
 // scroll wheel needs to be added not passively so it's cancelable, small helper hook to remember that
-function useScrollWheel({onScroll, capture}: {onScroll: (e) => void, capture: boolean}, ref: RefObject<HTMLElement>) {
+function useScrollWheel({onScroll, capture, disable}: {onScroll: (e) => void, capture: boolean, disable: boolean}, ref: RefObject<HTMLElement>) {
```
Comment (devongovett) [inline packages/@react-aria/numberfield/src/useNumberField.ts:393]: Should be `isDisabled`

Comment (snowystinger) [reply packages/@react-aria/numberfield/src/useNumberField.ts:393]: yeah, this is addressed in https://github.com/adobe/react-spectrum/pull/1654/files#diff-cce470e90db62eae114c603721371a2c5d91e0007b4d7fe4b1ec0d100226fb6bR22

Comment (snowystinger) [reply packages/@react-aria/numberfield/src/useNumberField.ts]: eh, i'll have merge conflicts anyways, addressed here as well

**Review by devongovett — APPROVED** [review]

**Pull request merged**